firmware.go

```diff
@@ -112,7 +112,7 @@ func stringFromReader(reader *bufio.Reader, file string, desc string) *string {
 		_log(nil, "Failed to read %v from %v: %v\n", desc, file, err)
 		return nil
 	}
-	str = strings.Trim(str, string(0))
+	str = strings.Trim(str, "0")
```

Comment:
This modification changes the code's behavior and breaks parsing firmware files ("0" != "\x00"). I, however, agree that string(0) is not an elegant solution and should be replaced ;)

Reply:
Removed my changes there